File: v2/java_sample/payloads/SigningUrl.java
package payloads;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.json.JSONObject;

public class SigningUrl {
	private int id;
	private String recipient_email;
	private String redirect_url = "";
	private boolean allow_decline = false;

	public SigningUrl(int id, String recipient_email)
	{
		this.id = id;
		this.recipient_email = recipient_email;
	}
	
	public void set_redirect_url(String val)
	{
		this.redirect_url  = val;
	}
	
	public void set_allow_decline(boolean val)
	{
		this.allow_decline   = val;
	}
	
	public JSONObject toJson() 
	{
		JSONObject obj = new JSONObject();

		obj.put("recipient_email", this.recipient_email);
		obj.put("redirect_url", this.redirect_url);
		obj.put("allow_decline", this.allow_decline);
		return obj;
	}

	public CloseableHttpResponse fetch_embedded_signing_url_for_embedded_sending(String base_url, String api_token) throws Exception 
	{
		String string = String.format("v2/rs/embedded/%d/signing/url", this.id);
        HttpPost post = new HttpPost(base_url + string);
        post.addHeader("content-type", "application/json");
        post.addHeader("Authorization", api_token);
        post.setEntity(new StringEntity(this.toJson().toString()));

        CloseableHttpClient httpClient = HttpClients.createDefault();
        CloseableHttpResponse response = httpClient.execute(post);

        return response;
    }
	
	public CloseableHttpResponse fetch_embedded_signing_url_for_envelopes(String base_url, String api_token) throws Exception 
	{
		String string = String.format("v2/rs/envelope/%d/signing/url", this.id);
        HttpPost post = new HttpPost(base_url + string);
        post.addHeader("content-type", "application/json");
        post.addHeader("Authorization", api_token);
        post.setEntity(new StringEntity(this.toJson().toString()));

        CloseableHttpClient httpClient = HttpClients.createDefault();
        CloseableHttpResponse response = httpClient.execute(post);

        return response;
    }
	
	public CloseableHttpResponse fetch_embedded_signing_url_for_templates(String base_url, String api_token) throws Exception 
	{
		String string = String.format("v2/template/rs/%d/signing/url", this.id);
        HttpPost post = new HttpPost(base_url + string);
        post.addHeader("content-type", "application/json");
        post.addHeader("Authorization", api_token);
        post.setEntity(new StringEntity(this.toJson().toString()));

        CloseableHttpClient httpClient = HttpClients.createDefault();
        CloseableHttpResponse response = httpClient.execute(post);

        return response;
    }
}

File: v2/node_sample/payload/SigningUrl.js
var unirest = require('unirest');

class SigningUrl {
	constructor(id, recipient_email, allow_decline=false, redirect_url='')
	{
		this.id = id;
		this.recipient_email = recipient_email;
		this.allow_decline = allow_decline;
		this.redirect_url = redirect_url;
	}

	to_json()
	{
		return {"recipient_email": this.recipient_email, "allow_decline": this.allow_decline,
		"redirect_url": this.redirect_url}
	}

	fetch_embedded_signing_url_for_embedded_sending = function (base_url, api_token) {
        var payload = this.to_json();
        var url = base_url + `v2/rs/embedded/${this.id}/signing/url`;
        return new Promise((resolve, reject) => {
        unirest.post(url)
        .headers({'Authorization': api_token, 'Content-Type': 'application/json'})
        .send(payload)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }

    fetch_embedded_signing_url_for_envelopes = function (base_url, api_token) {
        var payload = this.to_json();
        var url = base_url + `v2/rs/envelope/${this.id}/signing/url`;
        return new Promise((resolve, reject) => {
        unirest.post(url)
        .headers({'Authorization': api_token, 'Content-Type': 'application/json'})
        .send(payload)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }

    fetch_embedded_signing_url_for_templates = function (base_url, api_token) {
        var payload = this.to_json();
        var url = base_url + `v2/template/rs/${this.id}/signing/url`;
        return new Promise((resolve, reject) => {
        unirest.post(url)
        .headers({'Authorization': api_token, 'Content-Type': 'application/json'})
        .send(payload)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }
}

module.exports = SigningUrl